benches/fastx.rs

```diff
@@ -0,0 +1,189 @@
+#![feature(test)]
+
+extern crate test;
+
+use bio::io::{fasta, fastx};
+use rand::distributions::Alphanumeric;
+use rand::rngs::StdRng;
+use rand::Rng;
+use rand::SeedableRng;
+use std::io;
+use test::{black_box, Bencher};
+
+const BASES: &[u8] = b"ACTG";
+const ID_LEN: usize = 10;
+const DESC_LEN: usize = 20;
+const SEQ_LEN: usize = 100;
+const FASTA_SIZE: usize = 100;
+
+const ITERS: usize = 4000;
+
+fn gen_random_fasta() -> io::Result<Vec<u8>> {
+    let mut raw_writer = Vec::new();
+    {
+        let mut w = fasta::Writer::new(&mut raw_writer);
+        for _ in 0..FASTA_SIZE {
+            let id: String = StdRng::seed_from_u64(42)
+                .sample_iter(&Alphanumeric)
+                .take(ID_LEN)
+                .map(char::from)
+                .collect();
+
+            let desc: String = StdRng::seed_from_u64(4)
+                .sample_iter(&Alphanumeric)
+                .take(DESC_LEN)
+                .map(char::from)
+                .collect();
+
+            let seq: Vec<u8> = (0..SEQ_LEN)
+                .map(|_| {
+                    let idx = StdRng::seed_from_u64(65).gen_range(0..BASES.len());
+                    BASES[idx]
+                })
+                .collect();
+
+            w.write_record(&fasta::Record::with_attrs(&id, Some(&desc), &seq))?;
+        }
```

Comment:
I propose an important simplification of this code:

```rust
    let mut rng = StdRng::seed_from_u64(42);
    let mut raw_writer = Vec::new();
    {
        let mut w = fasta::Writer::new(&mut raw_writer);
        for _ in 0..FASTA_SIZE {
            let id: String = (0..ID_LEN)
                .map(|_| char::from(rng.sample(&Alphanumeric)))
                .collect();

            let desc: String = (0..DESC_LEN)
                .map(|_| char::from(rng.sample(&Alphanumeric)))
                .collect();

            let seq: Vec<u8> = (0..SEQ_LEN)
                .map(|_| {
                    let idx = rng.gen_range(0..BASES.len());
                    BASES[idx]
                })
                .collect();

            w.write_record(&fasta::Record::with_attrs(&id, Some(&desc), &seq))?;
        }
 ```
 Another advantage, of this code id, desc and sequence is different between each read, but file produce is same between run.